recipes/natural_language_processing/summarizer/app/summarizer.py

```diff
@@ -44,14 +45,30 @@ def chunk_text(text):
         text_list = text_list[chunk_size:]
     return chunks
 
+def read_file(file):
+    file_type = file.type
+    
+    if file_type == "application/pdf":
+        with open("tmp_pdf", "wb") as f:
```

Comment:
Shouldnt this use the file_name:
```suggestion
        with open(file_name, "wb") as f:
```
Also I am curious why we need to `f.write` before we can call it in with the loader

Reply:
I don't love that implementation either, happy to see if we can get around it. But for now the PDF loader we're using seems to want to read in a file from disk. So we write the streamlit file object to disk temporarily so we can then read it using the PDF loader.

Reply:
I see, that makes sense. We can iterate as we go but that is more than acceptable for a first pass, I was just wondering what the effect of this step was.